tests/kernel/timer/timer_api/src/main.c

```diff
@@ -270,10 +277,14 @@ void test_timer_periodicity(void)
 		 * one requested, as the kernel uses the ticks to manage
 		 * time. The actual perioid will be equal to [tick time]
 		 * multiplied by k_ms_to_ticks_ceil32(PERIOD).
+		 *
+		 * In the case of inexact conversion the delta will
+		 * occasionally be one less than the expected number.
 		 */
-		TIMER_ASSERT(WITHIN_ERROR(delta,
-				k_ticks_to_ms_floor64(k_ms_to_ticks_ceil32(PERIOD)), 1),
-				&periodicity_timer);
+		TIMER_ASSERT(WITHIN_ERROR(delta, period_ms, 1)
+			     || (INEXACT_MS_CONVERT
+				 && (delta == period_ms - 1)),
```

Comment:
Similar deal: it used to allow slop, now it doesn't.  We should do the same thing with a selectable offset so we don't end up accidentally testing different behavior on different systems.